cudapoa/src/batch.cpp

```diff
@@ -18,25 +18,43 @@ namespace claragenomics
 namespace cudapoa
 {
 
-std::unique_ptr<Batch> create_batch(int32_t max_sequences_per_poa,
-                                    int32_t device_id,
+std::unique_ptr<Batch> create_batch(int32_t device_id,
                                     cudaStream_t stream,
                                     size_t max_mem,
                                     int8_t output_mask,
+                                    BatchSize batch_size,
                                     int16_t gap_score,
                                     int16_t mismatch_score,
                                     int16_t match_score,
                                     bool cuda_banded_alignment)
 {
-    return std::make_unique<CudapoaBatch>(max_sequences_per_poa,
-                                          device_id,
-                                          stream,
-                                          max_mem,
-                                          output_mask,
-                                          gap_score,
-                                          mismatch_score,
-                                          match_score,
-                                          cuda_banded_alignment);
+    // a decision flag to determine proper type definition for ScoreT
+    const bool use_32_bit_for_ScoreT = use32bitInt(batch_size.max_nodes_per_window, match_score);
```

Comment:
since this file is using `use32bitInt` from `cudapoa_kernels.cuh`, let's explicitly include that header into this file

Reply:
done.